trl/trainer/utils.py

```diff
@@ -214,6 +214,89 @@ def torch_call(self, examples: List[Union[List[int], Any, Dict[str, Any]]]) -> D
         return batch
 
 
+@dataclass
+class DataCollatorForChatML:
+    """
+    Data collator for ChatML format datasets.
+    """
+
+    tokenizer: PreTrainedTokenizerBase
+    ignore_index: int = -100
+    max_length: int = None
+
+    def __post_init__(self):
+        if self.tokenizer.pad_token_id is None:
+            raise ValueError("The tokenizer does not have a pad token. Please set `pad_token_id` in the tokenizer.")
+        if self.max_length is None:
+            self.max_length = self.tokenizer.model_max_length
```

Comment:
Some models have `model_max_length` set to the largest Python `int` value, so for those cases we need a sensible default value + a warning